client/client.go

```diff
@@ -31,11 +31,106 @@ import (
 	"github.com/openpubkey/openpubkey/gq"
 	"github.com/openpubkey/openpubkey/pktoken"
 	"github.com/openpubkey/openpubkey/pktoken/clientinstance"
+	"github.com/openpubkey/openpubkey/util"
 )
 
 type OpkClient struct {
-	Op   OpenIdProvider
-	CosP *CosignerProvider
+	Op     OpenIdProvider
+	CosP   *CosignerProvider
+	signer crypto.Signer
+	alg    jwa.KeyAlgorithm
+	signGQ bool // Default is false
+}
+
+// ClientOpts contains options for constructing an OpkClient
+type ClientOpts func(o *OpkClient)
```

Comment:
In trying to decide what to name this I looked up how options are named in golang official crypto library to try to keep things consistent. 

As far as I can tell they use `Options` when the full variable name is just `Options` such as in [src/crypto/ed25519/ed25519.go](https://cs.opensource.google/go/go/+/master:src/crypto/ed25519/ed25519.go;l=273;bpv=1;bpt=1?q=options&ss=go%2Fgo)
`func VerifyWithOptions(publicKey PublicKey, message, sig []byte, opts *Options)`

but when the are specifying a prefix they use Opts rather than Option as in [src/crypto/crypto.go](https://cs.opensource.google/go/go/+/refs/tags/go1.21.6:src/crypto/crypto.go;l=198):
`Sign(rand io.Reader, digest []byte, opts SignerOpts) (signature []byte, err error`

I decided to stick with this pattern from go crypto.